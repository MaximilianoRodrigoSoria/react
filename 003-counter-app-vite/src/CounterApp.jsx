import { useState } from "react";
import { PropTypes } from "prop-types";


export const  CounterApp = ({value}) =>{
    
    const [counter, setCounter] = useState(value)

    const handleAdd = (  ) =>{
            setCounter(counter + 1);
        }

    const handleRemove = (  ) =>{
            setCounter((c) => c - 1);
        }

    const handleReset = (  ) =>
        {
            setCounter(value);
        }    

    return ( 
    <>
    <h1>CounterAPP</h1>
    <h2>Value: {counter} </h2>
    <button  onClick={handleAdd}>+1</button>
    <button  onClick={handleRemove}>-1</button>
    <button  onClick={handleReset}>Reset</button>
    </>)
}

;

CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}


export default CounterApp;

