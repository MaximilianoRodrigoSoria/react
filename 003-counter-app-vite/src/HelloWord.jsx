const app = ()=>(<h1>Hola mundo!!!</h1>);

export default app;

  