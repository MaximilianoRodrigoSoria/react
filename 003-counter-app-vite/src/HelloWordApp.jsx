const app = ({title})=>(<h1>Hola {title}!!!</h1>);

export default app;

  