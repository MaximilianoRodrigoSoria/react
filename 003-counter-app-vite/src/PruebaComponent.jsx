
const appPrueba = () => {
 return (
 <>
    <h2>Esto es una prueba de componente</h2>
    <p>Esto es un parrafo</p>
 </>)
};

export default appPrueba;