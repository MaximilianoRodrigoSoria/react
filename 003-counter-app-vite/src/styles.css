html,body {
    background-color: #21232A;
    color: white;
    font-family: Hervetica, Arial, Helvetica, sans-serif;
    font-family: 1.3rem;
    padding: 70px;

}

button{
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 10px;;
}