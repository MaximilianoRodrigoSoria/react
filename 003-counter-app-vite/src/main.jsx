import React from 'react';
import ReactDOM from 'react-dom/client';
import FirstApp from './FirstApp';
import HelloWord from './HelloWordApp'
import CounterApp from './CounterApp'

import './styles.css';

ReactDOM.createRoot(document.getElementById('root'))
.render(
  <React.StrictMode>

    <FirstApp 
          titulo = 'Esto es un titulo cualquiera'
          subtitulo = {123}
    >
    </FirstApp>
   {/* <CounterApp value={100}></CounterApp> */}
  </React.StrictMode>
)
;

















