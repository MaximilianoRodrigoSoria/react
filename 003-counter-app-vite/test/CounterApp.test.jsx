import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';



describe('Pruebas en <CounterAPP/>', () => {  
    const  initialValue = 100;
    
    test('Debe coincidir el Snapshot', () => {
        
        const { container } = render( <CounterApp value={ initialValue } /> );
        expect( container ).toMatchSnapshot();
    });   

    test('Debe de mostrar el valor inicial de 100 <CounterApp value={100}>', () => {
        
        render( <CounterApp value={ initialValue } /> );
        expect( screen.getByText(100) ).toBeTruthy();
        // expect( screen.getByRole('heading',{ level: 2}).innerHTML ).toContain('100')

    });

    test('Debe de incrementar con el botón +1', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('101') ).toBeTruthy();

    });

    test('Debe de decrementar con el botón -1', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('-1') );
        expect( screen.getByText('99') ).toBeTruthy();

    });

    test('Debe de funcionar el botón de reset', () => {
        
        render( <CounterApp value={ 355 } /> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect( screen.getByText( 355 ) ).toBeTruthy();

    });
    
});
