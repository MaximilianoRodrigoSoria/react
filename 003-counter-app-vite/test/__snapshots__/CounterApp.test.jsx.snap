// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterAPP/> Debe coincidir el Snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    id="+1"
  >
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
